src/EmbedIO/Utilities/IPParser.cs

```diff
@@ -0,0 +1,203 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace EmbedIO.Utilities
+{
+    /// <summary>
+    /// Provides standard methods to parse IP address strings.
+    /// </summary>
+    public static class IPParser
+    {
+        /// <summary>
+        /// Parses the specified IP address.
+        /// </summary>
+        /// <param name="address">The IP address.</param>
+        /// <returns>A collection of <see cref="IPAddress"/> parsed correctly from <paramref name="address"/>.</returns>
+        public static async Task<IEnumerable<IPAddress>> Parse(string address)
+        {
+            if (address == null)
+                return Enumerable.Empty<IPAddress>();
+
+            if (IPAddress.TryParse(address, out var ip))
+                return new List<IPAddress> { ip };
+
+            try
+            {
+                return await Dns.GetHostAddressesAsync(address).ConfigureAwait(false);
+            }
+            catch
+            {
+                // Ignore
+            }
+
+            if (IsCIDRNotation(address))
+                return ParseCIDRNotation(address);
+
+            return IsSimpleIPRange(address) ? TryParseSimpleIPRange(address) : Enumerable.Empty<IPAddress>();
+        }
+
+        /// <summary>
+        /// Determines whether the IP-range string is in CIDR notation.
+        /// </summary>
+        /// <param name="range">The IP-range string.</param>
+        /// <returns>
+        ///   <c>true</c> if the IP-range string is CIDR notation; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsCIDRNotation(string range)
+        {
+            if (string.IsNullOrWhiteSpace(range))
+                return false;
+
+            var parts = range.Split('/');
+            if (parts.Length != 2)
+                return false;
+
+            var prefix = parts[0];
+            var prefixLen = parts[1];
+
+            var prefixParts = prefix.Split('.');
+            if (prefixParts.Length != 4)
+                return false;
+
+            return byte.TryParse(prefixLen, out var len) && len >= 0 && len <= 32;
+        }
+
+        /// <summary>
+        /// Parse IP-range string in CIDR notation. For example "12.15.0.0/16".
+        /// </summary>
+        /// <param name="range">The IP-range string.</param>
+        /// <returns>A collection of <see cref="IPAddress"/> parsed correctly from <paramref name="range"/>.</returns>
+        public static IEnumerable<IPAddress> ParseCIDRNotation(string range)
```

Comment:
Same as above: `ParseCidrNotation` would be a better name for this method.

Reply:
I already changed the method's name.